common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -484,6 +575,70 @@ public void putValueInSuccessEvent(final @NonNull ApiEndEvent event, final @NonN
                 });
     }
 
+    /**
+     * Get the intent for the broker Account Transfer request
+     *
+     * @param parameters a {@link AccountTransferTokenCommandParameters}
+     * @return an {@link Intent} for initiating Broker Account Transfer interactive activity.
+     */
+    @NonNull
+    private Intent getBrokerAuthorizationIntentForAccountTransfer(
+            final @NonNull AccountTransferTokenCommandParameters parameters) throws BaseException {
+        return getBrokerOperationExecutor().execute(parameters,
+                new BrokerOperation<Intent>() {
+                    private String negotiatedBrokerProtocolVersion;
+
+                    @Override
+                    public void performPrerequisites(final @NonNull IIpcStrategy strategy) throws BaseException {
+                        verifyTokenParametersAreSupported(parameters);
+                        negotiatedBrokerProtocolVersion = hello(strategy, parameters.getRequiredBrokerProtocolVersion());
+                    }
+
+                    @Override
+                    public @NonNull
+                    BrokerOperationBundle getBundle() {
+                        return new BrokerOperationBundle(
+                                MSAL_GET_INTENT_FOR_ACCOUNT_TRANSFER_INTERACTIVE_REQUEST,
+                                mActiveBrokerPackageName,
+                                null);
```

Comment:
alternatively, you can create a bundle with a flag here.

on the broker side, if the flag exists, return the AccountTransfer activity class. 
otherwise, return AccountChooser.

If you do it this way, then the change on msal side would be minimal. you'll get a lot of things for free.

(you can pretty much reuse the existing acquire token interactive code in this class. The only difference would be in this getBundle() method)

Reply:
could you please remind me again what's the reason of opening a new ipc interface/new activity?

Reply:
No longer needed